lib/src/protocol_instance.rs

```diff
@@ -109,7 +109,7 @@ pub fn assemble_protocol_instance(
         transition: Transition {
             parentHash: header.parent_hash,
             blockHash: header.hash(),
-            stateRoot: input.taiko.l1_header.state_root,
+            stateRoot: header.state_root,
```

Comment:
Is transition's stateRoot by design L1 or L2??

Reply:
Oh yes it's the L2 one!